R/scale-type.R

```diff
@@ -25,7 +24,10 @@ find_scale <- function(aes, x, env = parent.frame()) {
 # Look for object first in parent environment and if not found, then in
 # ggplot2 namespace environment.  This makes it possible to override default
 # scales by setting them in the parent environment.
-find_global <- function(name, env, mode = "any") {
+find_global <- function(name, env, mode = "any", theme_ = theme()) {
+  if (!is.null(theme_[["default.scales"]][[name]]) && mode(theme_[["default.scales"]][[name]]) == mode) {
+    return(theme_[["default.scales"]][[name]])
```

Comment:
Theme elements should generally be retrieved with `calc_element()`, not by direct lookup.

Reply:
Also, you're doing the same lookup three times. I think it makes more sense to do it once and assign to a temporary variable.

Reply:
Ah thanks, not poked around in internals much so I'm likely to miss things like that.